cmd/libsnap-confine-private/facts.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SNAP_CONFINE_FACTS_H
+#define SNAP_CONFINE_FACTS_H
+
+#include <stdbool.h>
+#include <stddef.h>
+
+/* Facts are stored as multi-line strings using key=value syntax */
+
+/**
+ * Directory where facts are stored by snapd.
+ *
+ * The directory *may* be absent.
+**/
+#define SC_FACT_DIR "/var/lib/snapd/facts"
+
+/**
+ * Load facts from a given file.
+ *
+ * The file must contain key=value facts, one per line. The file may be absent,
+ * it is equivalent to an empty file. Facts are limited to, at most, 16KB of
+ * data.
+ *
+ * The return value must by released by calling free(3).
+ **/
+char *sc_load_facts(const char *fname);
```

Comment:
Since this function is obviously meant to be reused for different fact files, would it make sense to take max size as an argument rather then hardcode it to 16KB in the functon? Size of 0 could indicate a default value of 16KB.

Reply:
I'd YAGNI it because it's a sane default for now and iff we ever need to use it for other things this is something we can change. Still even if we had access to all of core's configuration it would fit in 16KB

Reply:
Fair enough, easy to change afterwards if we need to.